### PR TITLE
The mkldnn batch norm supports other data format

### PR DESCRIPTION
**Pull request by mozga-intel**
The batch norm operator will be support the other type of the input data.

### CODE REVIEW

File: paddle/fluid/operators/batch_norm_mkldnn_op.cc [hunk 1]
```diff
@@ -113,9 +113,16 @@ class BatchNormMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
...
     // create mkldnn memory from input x tensor
-    auto src_memory =
-        memory({{{src_tz}, memory::data_type::f32, x->format()}, mkldnn_engine},
-               to_void_cast(x_data));
+    mkldnn::memory::format input_format = x->format();
+    if (src_tz.size() == 1) {
+      input_format = mkldnn::memory::format::x;
+    } else if (src_tz.size() == 2) {
+      input_format = mkldnn::memory::format::nc;
+    }
```
Comment (luotao1) [inline paddle/fluid/operators/batch_norm_mkldnn_op.cc]: You can use a function to package following codes, since they are duplicated three times in this file.
```
if (src_tz.size() == 1) {
  input_format = mkldnn::memory::format::x;
    } else if (src_tz.size() == 2) {
      input_format = mkldnn::memory::format::nc;
    }
```
And I see there is an inline function `MKLDNNFormat MKLDNNFormatForSize` in https://github.com/PaddlePaddle/Paddle/pull/11544/files#diff-70d521f84c13ef539b9ef82a8886b693. Maybe you can use it as well. How about create a PR first to add this function?

Comment (mozga-intel) [reply paddle/fluid/operators/batch_norm_mkldnn_op.cc]: Sure, I have been changing it.

File: paddle/fluid/platform/mkldnn_helper.h [hunk 2]
```diff
@@ -237,5 +237,16 @@ class MKLDNNHandler {
   bool is_reusing_;
 };
 
+inline mkldnn::memory::format MKLDNNFormatForSize(
+    size_t dims_size, mkldnn::memory::format data_format) {
+  if (dims_size == 1) {
+    if (dims_size == 2) {
```
Comment (tensor-tang) [inline paddle/fluid/platform/mkldnn_helper.h]: Is that right? maybe forget else ?

```
dims_size == 1
              ? mkldnn::memory::format::x
              : dims_size == 2 ? mkldnn::memory::format::nc : default_format
```

Comment (luotao1) [reply paddle/fluid/platform/mkldnn_helper.h]: Could you reuse the code in  https://github.com/PaddlePaddle/Paddle/pull/11544/files#diff-70d521f84c13ef539b9ef82a8886b693 ?

File: paddle/fluid/platform/mkldnn_helper.h [hunk 3]
```diff
@@ -237,5 +237,15 @@ class MKLDNNHandler {
   bool is_reusing_;
 };
 
+inline mkldnn::memory::format MKLDNNFormatForSize(
```
Comment (tensor-tang) [inline paddle/fluid/platform/mkldnn_helper.h:240]: There is a same function at here

https://github.com/PaddlePaddle/Paddle/blob/066f537ad2f446da670e861159adbba26fd0c568/paddle/fluid/framework/data_layout_transform.h#L65-L70

Maybe you can directly remove the same one.

Comment (mozga-intel) [reply paddle/fluid/platform/mkldnn_helper.h:240]: Done.

**Review by luotao1 — APPROVED** [review]
LGTM
